### ISSUE TITLE
swagger-jaxrs2 creates not existing root endpoint when using subresources

### ISSUE THREAD

**Issue by Santobert** [swagger-api/swagger-core#4283]
When using subresources according to the Jersey documentation, Swagger also incorrectly creates a root endpoint for each function annotated with `@Path`.
Documentation: https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest3x/jaxrs-resources.html#d0e2323

Example:
```java
@Path("/item")
public class ItemResource {
    @Context UriInfo uriInfo;
 
    @Path("content")
    public ItemContentResource getItemContentResource() {
        return new ItemContentResource();
    }
 
    @GET
    @Produces("application/xml")
        public Item get() { ... }
    }
}
 
public class ItemContentResource {
 
    @GET
    public Response get() { ... }
 
    @PUT
    @Path("{version}")
    public void put(@PathParam("version") int version,
                    @Context HttpHeaders headers,
                    byte[] in) {
        ...
    }
}
```

In this case, Swagger would correctly create the `/item/content/{version}` endpoint, but also create the `/{version}` endpoint, which is wrong.

This issue also occurs with the Swagger Sample App:
https://github.com/swagger-api/swagger-samples/tree/2.0/java/java-jersey2-configfile

![grafik](https://user-images.githubusercontent.com/15657682/198551266-de153bd0-920b-4c0c-aebb-3297d0b2ab35.png)

## Setup
- Java 17
- Jersey 3.0.8
- Swagger Version 2.2.4
- `-jakarta` packages

## Related
- https://github.com/swagger-api/swagger-core/issues/3678
- https://groups.google.com/g/swagger-swaggersocket/c/zFBnUEPJDSw

**Comment by kaporten** [comment]
Still an issue in 2.2.8 as well. This entirely prevents upgrading to jaxrs2 for projects making heavy use of sub-resources.

**Comment by frantuma** [comment]
Closing as not reproducible, see [this test](https://github.com/swagger-api/swagger-core/blob/e9389e988e9954673e81ed71519552a8dac1996d/modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/ReaderTest.java#L3100). Please reopen if you're still experiencing issues

**Comment by frantuma** [comment]
Which subresource has no path? the [StatusSubResource](https://github.com/swagger-api/swagger-core/blob/e9389e988e9954673e81ed71519552a8dac1996d/modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/resources/StatusSubResource.java#L12) has the `@Path` annotation

**Comment by Santobert** [comment]
The subresource in the test you referenced. Can you reproduce the issue with the sample code? Tbh, I've only tried my own project this morning.

**Comment by frantuma** [comment]
Please see the referenced test branch with the test not reproducing the issue

**Comment by kaporten** [comment]
Chiming in to confirm that this issue is fixed on my project from version 2.2.9 :)
